### PR TITLE
Closes #561 Centralize App URL definition

### LINKED ISSUE #561: URL for the app API should be defined in one place.
The URL for reaching the API is defined in (at least) 3 different places.

`inc/functions/common.php`
`inc/classes/class-imagify.php`
`assets/js/pricing-modal.js`

This should be defined once and reused.

### PR DESCRIPTION
**Pull request by mostafa-hisham**
define app url and use it i PHP and js

### CODE REVIEW

File: inc/classes/class-imagify.php [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Imagify {
 	 *
 	 * @var string
 	 */
-	const API_ENDPOINT = 'https://app.imagify.io/api/';
+	const API_ENDPOINT = IMAGIFY_APP_DOMAIN.'/api/';
```
Comment (iCaspar) [inline inc/classes/class-imagify.php]:
```suggestion
	const API_ENDPOINT = IMAGIFY_APP_DOMAIN . '/api/';
```

File: inc/functions/common.php [hunk 2]
```diff
@@ -223,8 +223,8 @@ function imagify_can_optimize_custom_folders() {
  * @return string The URL.
  */
 function imagify_get_external_url( $target, $query_args = array() ) {
-	$site_url = 'https://imagify.io/';
-	$app_url  = 'https://app.imagify.io/#/';
+	$site_url = IMAGIFY_APP_DOMAIN.'/';
```
Comment (iCaspar) [inline inc/functions/common.php]:
```suggestion
	$site_url = IMAGIFY_APP_DOMAIN . '/';
```

File: inc/functions/common.php [hunk 3]
```diff
@@ -223,8 +223,8 @@ function imagify_can_optimize_custom_folders() {
...
+	$app_url  = IMAGIFY_APP_DOMAIN.'/#/';
```
Comment (iCaspar) [inline inc/functions/common.php]:
```suggestion
	$app_url  = IMAGIFY_APP_DOMAIN . '/#/';
```

**Review by iCaspar — CHANGES_REQUESTED** [review]
Looks good, except these spacings will trigger phpcs even after the 7.4 update.

File: inc/functions/common.php (hunk 3 above)
Comment (mostafa-hisham) [reply inc/functions/common.php]: fixed

File: inc/functions/common.php (hunk 2 above)
Comment (mostafa-hisham) [reply inc/functions/common.php]: fixed

File: inc/classes/class-imagify.php (hunk 1 above)
Comment (mostafa-hisham) [reply inc/classes/class-imagify.php]: this one generated an error in php cs
`Constant scalar expressions are not allowed when defining constants using the const keyword in PHP 5.5 or earlier.`
so I added new constant for API and used it in this class

https://github.com/wp-media/imagify-plugin/blob/5d82893ac318aae56da21bc3740469ec7dda4b17/imagify.php#L33

**Review by iCaspar — APPROVED** [review]

**Comment by iCaspar** [comment]
Ready for @wp-media/qa 

This will resolve the issue about the pricing modal closing in test envs. The app URI to test for various environments is now in `imagify.php` -- set and forget! 🤞

**Comment by piotrbak** [comment]
@iCaspar @mostafa-hisham What would you say about defining it like we do in WP Rocket?
https://github.com/wp-media/wp-rocket/blob/a94306216879962bdee8736fb93fd0f38b5de8da/wp-rocket.php#L60

It would allow us to define it in `wp-config.php` and make things even easier (switching between branches, etc.)

**Comment by mostafa-hisham** [comment]
@piotrbak
Great idea 
Now we check first if it is not defined on wp-config.php
please check my changes

**Comment by piotrbak** [comment]
@iCaspar @mostafa-hisham The JS change should make the payment modal to be closed automatically after the successful checkout, right?

If that's the case, then it's not working.

**Pull request merged**